marklogic-data-hub/src/main/java/com/marklogic/hub/impl/InstallInfoImpl.java

```diff
@@ -0,0 +1,295 @@
+                areForestsExistent(DatabaseKind.FINAL) &&
+                areForestsExistent(DatabaseKind.TRACE) &&
+                areForestsExistent(DatabaseKind.JOB)
+        );
+
+        return (appserversOk && dbsOk && forestsOk);
+    }
+
+    @Override public String toString() {
+        return "\n" +
+            "Checking MarkLogic Installation:\n" +
+            "\tAppServers:\n" +
+            "\t\tStaging: " + (isAppServerExistent(DatabaseKind.STAGING) ? "exists" : "MISSING") + "\n" +
+            "\t\tFinal:   " + (isAppServerExistent(DatabaseKind.FINAL) ? "exists" : "MISSING") + "\n" +
+            "\t\tTrace:   " + (isAppServerExistent(DatabaseKind.TRACE) ? "exists" : "MISSING") + "\n" +
+            "\t\tJob:     " + (isAppServerExistent(DatabaseKind.JOB) ? "exists" : "MISSING") + "\n" +
+            "\tDatabases:\n" +
+            "\t\tStaging: " + (isDbExistent(DatabaseKind.STAGING) ? "exists" : "MISSING") + "\n" +
+            "\t\tFinal:   " + (isDbExistent(DatabaseKind.FINAL) ? "exists" : "MISSING") + "\n" +
+            "\t\tTrace:   " + (isDbExistent(DatabaseKind.TRACE) ? "exists" : "MISSING") + "\n" +
+            "\t\tJob:     " + (isDbExistent(DatabaseKind.JOB) ? "exists" : "MISSING") + "\n" +
+            "\tDatabases Indexes:\n" +
+            "\t\tStaging Triples Index : " + (isTripleIndexOn(DatabaseKind.STAGING) ? "exists" : "MISSING") + "\n" +
+            "\t\tStaging Collection Lexicon : " + (isCollectionLexiconOn(DatabaseKind.STAGING) ? "exists" : "MISSING") + "\n" +
+            "\t\tFinal Triples Index : " + (isTripleIndexOn(DatabaseKind.FINAL) ? "exists" : "MISSING") + "\n" +
+            "\t\tFinal Collection Lexicon : " + (isCollectionLexiconOn(DatabaseKind.FINAL) ? "exists" : "MISSING") + "\n" +
+            "\tForests\n" +
+            "\t\tStaging: " + (areForestsExistent(DatabaseKind.STAGING) ? "exists" : "MISSING") + "\n" +
+            "\t\tFinal:   " + (areForestsExistent(DatabaseKind.FINAL) ? "exists" : "MISSING") + "\n" +
+            "\t\tTrace:   " + (areForestsExistent(DatabaseKind.TRACE) ? "exists" : "MISSING") + "\n" +
+            "\t\tJob:     " + (areForestsExistent(DatabaseKind.JOB) ? "exists" : "MISSING") + "\n" +
+            "\n\n" +
+            "OVERAL RESULT: " + (isInstalled() ? "INSTALLED" : "NOT INSTALLED") + "\n";
+    }
+
+    @Override public boolean isAppServerExistent(DatabaseKind kind) {
+        boolean exists = false;
+        switch (kind) {
+            case STAGING:
+                exists = stagingAppServerExists;
+                break;
+            case FINAL:
+                exists = finalAppServerExists;
+                break;
+            case JOB:
+                exists = jobAppServerExists;
+                break;
+            case TRACE:
+                exists = traceAppServerExists;
+                break;
+            default:
+                throw new InvalidDBOperationError(kind, "test appserver existence");
+        }
+        return exists;
+    }
```

Comment:
begs the question whether the databases should all be in a HashMap<DatabaseKind,DatabasClient>  of course that can be a future refactor too

Reply:
Ideally attached and managed from in hubconfig, as well.